bee-common/bee-common/src/lib.rs

```diff
@@ -6,9 +6,5 @@
 
 #![warn(missing_docs)]
```

Comment:
I am in favor of denying warnings also. I know the CI would keep us from merging, but making it explicit in the code can be regarded positively by reviewers and users.

Reply:
@thibault-martinez Are you ok with this? I'm happy to switch it to `deny`.

Reply:
Of course :)

Reply:
But I'm not sure that's what Alex was suggesting

Reply:
I think he meant adding `#![deny(warnings)]`

Reply:
C'est correct!

Reply:
That was what I was suggesting also... Now I'm curious as to what Thibault thought I was suggesting!